docs/platform-external/use-case-oci-upi.md

```diff
@@ -0,0 +1,1195 @@
+            value: "6443"
+          - name: KUBERNETES_SERVICE_PORT_HTTPS
+            value: "6443"
+EOF
+
+# Merge required objects for pod template
+./yq eval-all '. as $item ireduce ({}; . *+ $item)' oci-cloud-controller-manager.yaml oci-cloud-controller-manager-ds_patch1.yaml > oci-cloud-controller-manager-ds_patched1.yaml
+
+# Merge required objects for containers
+./yq eval-all '.spec.template.spec.containers[] as $item ireduce ({}; . *+ $item)' oci-cloud-controller-manager-ds_patched1.yaml ./oci-cloud-controller-manager-ds_patch2.yaml > ./oci-cloud-controller-manager-ds_patched2.yaml
+
+# merge patched files
+./yq eval-all '.spec.template.spec.containers[] *= load("./oci-cloud-controller-manager-ds_patched2.yaml")' oci-cloud-controller-manager-ds_patched1.yaml > ${INSTALL_DIR}/manifests/oci-01-ccm-02-daemonset.yaml
+```
+
+The following CCM manifests files must be created in the installation `manifests/` directory:
+
+```bash
+$ tree $INSTALL_DIR/manifests/
+[...]
+├── oci-00-namespace.yaml
+├── oci-01-ccm-00-secret.yaml
+├── oci-01-ccm-01-rbac_0.yml
+├── oci-01-ccm-01-rbac_1.yml
+├── oci-01-ccm-01-rbac_2.yml
+├── oci-01-ccm-02-daemonset.yaml
+[...]
+```
+
+#### Create custom manifests for Kubelet
+
+This section describes the steps to create MachineConfig for kubeket to set the unique ID of the instance that an external provider used to identify a specific node.
+
+The Provider ID must be set dynamically for each node. The steps below describes how to create a MachineConfig manifest to setup a systemd unit to create a kubelet configuration discoverying the Provider ID in OCI by querying the [Instance Metadata Service (IMDS)](https://docs.oracle.com/en-us/iaas/Content/Compute/Tasks/gettingmetadata.htm).
+
+- Create the butane file:
+
+```bash
+function create_machineconfig_kubelet() {
+    local node_role=$1
+    cat << EOF > ./mc-kubelet-$node_role.bu
+# Butane file to setup kubelet for node role $node_role
+variant: openshift
+version: 4.12.0
+metadata:
+  name: 00-$node_role-kubelet-providerid
+  labels:
+    machineconfiguration.openshift.io/role: $node_role
+storage:
+  files:
+  - mode: 0755
+    path: "/usr/local/bin/kubelet-providerid"
+    contents:
+      inline: |
+        #!/bin/bash
+        set -e -o pipefail
+        NODECONF=/etc/systemd/system/kubelet.service.d/20-providerid.conf
+        if [ -e "\${NODECONF}" ]; then
+            echo "Not replacing existing \${NODECONF}"
+            exit 0
+        fi
+
+        PROVIDERID=\$(curl -H "Authorization: Bearer Oracle" -sL http://169.254.169.254/opc/v2/instance/ | jq -r .id);
+
+        cat > "\${NODECONF}" <<EOF
+        [Service]
+        Environment="KUBELET_PROVIDERID=providerName://\${PROVIDERID}"
+        EOF
+systemd:
+  units:
+  - name: kubelet-providerid.service
+    enabled: true
+    contents: |
+      [Unit]
+      Description=Fetch kubelet provider id from Metadata
+      Wants=afterburn.service
+      After=afterburn.service
```

Comment:
FYI, I managed to make it work without afterburn, I removed it because the unit was in error.

Reply:
@adriengentil so did you removed only the `After` requirement for the unit?

Reply:
all references to it: `Wants`, `After` and `EnvironmentFile`

Reply:
Perfect, thanks! afterburn dependencies have been removed.